src/components/component/toolbar.tsx

```diff
@@ -0,0 +1,201 @@
+      restartRequestMessage,
+    } = this.props;
+
+    const isStartEnabled =
+      component?.status === ComponentStatus.StoppedComponent &&
+      startRequestStatus !== RequestState.IN_PROGRESS;
+
+    const isStopEnabled =
+      component?.status !== ComponentStatus.StoppedComponent &&
+      component?.replicaList?.length > 0 &&
+      stopRequestStatus !== RequestState.IN_PROGRESS;
+
+    const isRestartEnabled =
+      component?.status === ComponentStatus.ConsistentComponent &&
+      component?.replicaList?.length > 0 &&
+      restartRequestStatus !== RequestState.IN_PROGRESS;
+
+    const restartInProgress =
+      restartRequestStatus === RequestState.IN_PROGRESS ||
+      component?.status === ComponentStatus.ComponentReconciling ||
+      component?.status === ComponentStatus.ComponentRestarting;
+
+    return (
+      <div className="grid grid--gap-small">
+        <div className="grid grid--gap-small grid--auto-columns">
+          {this.props.startEnabled && (
+            <Button onClick={this.doStartComponent} disabled={!isStartEnabled}>
+              Start
+            </Button>
+          )}
+          {this.props.stopEnabled && (
+            <Button onClick={this.doStopComponent} disabled={!isStopEnabled}>
+              Stop
+            </Button>
+          )}
+          <Button
+            onClick={this.doRestartComponent}
+            disabled={!isRestartEnabled}
+            variant="outlined"
+          >
+            Restart
+          </Button>
+          {restartInProgress && <CircularProgress size={32} />}
+        </div>
+        {startRequestMessage && <Typography>{startRequestMessage}</Typography>}
+        {stopRequestMessage && <Typography>{stopRequestMessage}</Typography>}
+        {restartRequestMessage && (
+          <Typography>{restartRequestMessage}</Typography>
+        )}
+      </div>
+    );
+  }
+}
+
+function mapStateToProps(state: RootState): ToolbarState {
+  return {
+    startRequestStatus: componentStartState.getStartRequestStatus(state),
+    startRequestMessage: componentStartState.getStartRequestError(state),
+    stopRequestStatus: componentStopState.getStopRequestStatus(state),
+    stopRequestMessage: componentStopState.getStopRequestError(state),
+    restartRequestStatus: componentRestartState.getRestartRequestStatus(state),
+    restartRequestMessage: componentRestartState.getRestartRequestError(state),
+  };
+}
+
+function mapDispatchToProps(dispatch: Dispatch) {
+  return {
+    startComponent: (appName, envName, componentName) =>
+      dispatch(
+        componentActions.start.startRequest(appName, envName, componentName)
+      ),
+    stopComponent: (appName, envName, componentName) =>
+      dispatch(
+        componentActions.stop.stopRequest(appName, envName, componentName)
+      ),
+    restartComponent: (appName, envName, componentName) =>
+      dispatch(
+        componentActions.restart.restartRequest(appName, envName, componentName)
+      ),
+  };
+}
```

Comment:
add function return type
```suggestion
function mapDispatchToProps(dispatch: Dispatch): ToolbarAction  {
  return {
    startComponent: (appName, envName, componentName) =>
      dispatch(
        componentActions.start.startRequest(appName, envName, componentName)
      ),
    stopComponent: (appName, envName, componentName) =>
      dispatch(
        componentActions.stop.stopRequest(appName, envName, componentName)
      ),
    restartComponent: (appName, envName, componentName) =>
      dispatch(
        componentActions.restart.restartRequest(appName, envName, componentName)
      ),
  };
}
```